chapters/en/slides/module3/module3_02.Rmd

```diff
@@ -0,0 +1,308 @@
+and then count plots fully in one bin or another,
+we can create bins that are centered
+on the data and then add them together.
+
+When we center bins on the data,
+we also usually have a round shape 
+instead of a square one,
+which gives the impression of a smoother distribution
+that is more conducive to communicating the shape of the data distribution.
+
+The round bins (also called kernels)
+are then added together as in the gif in this slide,
+and sum up to an overall distribution line.
+Formally,
+this is called a Kernel Density Estimate (KDE),
+or just a density plot.
+
+---
+
+## The movies dataset
+
+```{python}
+import pandas as pd
+
+
+movies = pd.read_csv('movies.csv')
+movies
+```
+
+Notes:
+We will be using the movies dataset again,
+let's see if we can make a plot
+that is easier to interpret
+than the layered histograms
+from the end of the last slide deck.
+
+---
+
+## Creating a density plot in Altair required two step
+
+```{python}
+import altair as alt
+
+(alt.Chart(movies).transform_density(
+    'runtime',
+    as_=['runtime', 'density'])  # Give a name to the KDE values, which we can use when plotting
+ .mark_area().encode(
+    x='runtime',
+    y='density:Q'))
+```
+
+Notes:
+To visualize a density plot,
+we need to do two things:
+
+1. Calculate the KDE (adding the individual kernels and summing them together)
+2. Plot a line or area mark for the newly calculated KDE values.
+
+In Altair,
+these operations are done in two explicit steps,
+using `transform_density` for the calculation.
+We could also have calculated this as a new column in our pandas data frame
+and then plotted that column,
+but it is convenient to do both the steps in Altair.
+
+The new variable we create is called `'density'`,
+and since this is not part of the pandas data frame,
+Altair cannot ask pandas which data type it is.
+Therefore,
+we need to add `':Q'` to indicate that the density has quantitative (numeric) values.
+
+---
+
+## Creating a grouped density plot requires an explicit density grouping
+
+```{python}
+(alt.Chart(movies).transform_density(
+     'runtime',
+     groupby=['genre'],
+     as_=['runtime', 'density'])
+ .mark_area().encode(
+     x='runtime',
+     y='density:Q',
+     color='genre'))
+```
+
+Notes:
+If we want to split and color the densities
+by a categorical variable,
+we need to add an explicit `groupby` column
+when calculating the density.
+This computes one density 
+per color variable (`genre` in this case),
+that will then be colored accordingly
+since we used the same variable for the color argument.
+
+In this plot,
+it is quite easy to see the relationship between movie lengths of different genres.
+The history genre has the longest running movies
+as well as the most variation
+whereas most animation moves are shorter
+and with less spread within the genre.
+
+There are still a few things we could tweak to improve this plot!
+
+---
+
+## Creating smoother density plots with more calculation steps
```

Comment:
I think a slide needs to be devoted to explaining what is going on here, at least at a conceptual level. Maybe an animation like you had in slide 3?

Reply:
Agree that an animation would be great but don't think that I have seen one. I might make a few plots instead. Ideally, I would not talk about this at all. I wish the default value for the steps was higher, I'm going to open a vegalite issue